File: DN - DotNet FSE Mandatory hands-on/Design Patterns and Principles/Exercise 1 Implementing the Singleton Pattern/src/Logger.java
public class Logger {

    // Single object
    private static Logger instance;

    // Private constructor
    private Logger() {
        System.out.println("Logger Created");
    }

    // Static method
    public static Logger getInstance() {

        if(instance == null){
            instance = new Logger();
        }

        return instance;
    }

    public void log(String message){
        System.out.println("LOG : " + message);
    }

}

File: DN - DotNet FSE Mandatory hands-on/Design Patterns and Principles/Exercise 1 Implementing the Singleton Pattern/Logger.cs
using System;

namespace SingletonPatternExample
{
    public class Logger
    {
        // Single instance
        private static Logger instance;

        // Private constructor
        private Logger()
        {
        }

        // Public method to access instance
        public static Logger GetInstance()
        {
            if (instance == null)
            {
                instance = new Logger();
            }

            return instance;
        }

        public void Log(string message)
        {
            Console.WriteLine("Log: " + message);
        }
    }
}